src/controllers/ioSignController.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * This controller handles the IO_SIGN requests from the
+ * app by forwarding the call to the API system.
+ */
+
+import * as express from "express";
+import * as TE from "fp-ts/TaskEither";
+import * as E from "fp-ts/Either";
+import { sequenceS } from "fp-ts/lib/Apply";
+
+import {
+  IResponseErrorForbiddenNotAuthorized,
+  IResponseErrorInternal,
+  IResponseErrorNotFound,
+  IResponseErrorValidation,
+  IResponseSuccessJson,
+  ResponseErrorInternal,
+  ResponseErrorValidation
+} from "@pagopa/ts-commons/lib/responses";
+
+import IoSignService from "src/services/ioSignService";
+import { pipe } from "fp-ts/lib/function";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { errorsToReadableMessages } from "@pagopa/ts-commons/lib/reporters";
+import { SignerDetailView } from "../../generated/io-sign-api/SignerDetailView";
+
+import { FilledDocumentDetailView } from "../../generated/io-sign-api/FilledDocumentDetailView";
+
+import { CreateFilledDocument } from "../../generated/io-sign/CreateFilledDocument";
+
+import { User, withUserFromRequest } from "../types/user";
+import ProfileService from "../services/profileService";
+
+import { profileWithEmailValidatedOrError } from "../utils/profile";
+
+export const retrieveSignerId = (ioSignService: IoSignService, user: User) =>
```

Comment:
I've thought about this a little bit more and: you don't need this method (`retrieveSignerId`) at all since all the errors are handled in the `ioSignService.getSignerByFiscalCode` method.

Just use that one in `createFilledDocument`